common-theurgy.lic

```diff
@@ -155,8 +155,12 @@ module DRCTH
   end
 
   def sprinkle?(item, target)
-    result = DRC.bput("sprinkle #{item} on #{target}", 'You sprinkle', 'Sprinkle what', 'What were you referring to')
-    result =~ /'You sprinkle'/
+    case DRC.bput("sprinkle #{item} on #{target}", 'You sprinkle', 'Sprinkle (what|that)', 'What were you referring to')
+    when /^You sprinkle/
```

Comment:
This is one way to fix the problem. However, I think the original code was cleaner.

What do you think about keeping the original approach (using `=~`), but just fixing the bug in the regular expression (dropping the `''`, optionally adding the `^`)?

Reply:
Sorry, I was trying to wrap my head around it, and whilst I accept it's neater to have the two-liner... it's taken hours to wrap my head around what it does, or how to use it. @asechrest has very helpfully worked it out with me, and we think we have a good handle on it now, but honestly, that's iffy on my part.

My inclination is that all our predicates should be, well, predicates and return explicit true false. Even getting to where I can test this has been an effort. I'll make it how you're suggesting if you're set on it, but honestly, I prefer something more easily understood by someone with a lower skill level. Either the case bput, or even making it an `if` that explicitly returns true, would be much more accessible for folks coming into the project. 

Does that make sense?

Reply:
I see your point re: explicit true/false return values. I'm not generally a fan of relying on ambiguous truthiness.

Clarity > conciseness, but I wonder if there's a way to get both. What do you think about this?

```
def sprinkle?(item, target)
    result = DRC.bput("sprinkle #{item} on #{target}", 'You sprinkle', 'Sprinkle what', 'What were you referring to')
    result == 'You sprinkle'
end
```

Reply:
Let me test it out a bit today. That makes a lot more sense to me than =~

Reply:
Tests right. Works for me.